### ISSUE TITLE
Bug: Healthcheck going between healthy and unhealthy constantly

### ISSUE THREAD

**Issue by Yaikava** [qdm12/gluetun#1655]
### Is this urgent?

No

### Host OS

dietpi/debian 11

### CPU arch

aarch64

### VPN service provider

IVPN

### What are you using to run the container

docker-compose

### What is the version of Gluetun

2023-06-01T16:49:06.663Z (commit 943943e)

### What's the problem 🤔

Running on raspberry pi 4B.
VPN seems to be going unhealthy often for no reason.
Any idea why?

### Share your logs

```plain text
vpn  | 2023-06-05T01:34:24+02:00 INFO [healthcheck] unhealthy: dialing: dial tcp4 1.1.1.1:443: i/o timeout
vpn  | 2023-06-05T01:34:32+02:00 INFO [healthcheck] program has been unhealthy for 6s: restarting VPN (see https://github.com/qdm12/gluetun/wiki/Healthcheck)
vpn  | 2023-06-05T01:34:32+02:00 INFO [vpn] stopping
vpn  | 2023-06-05T01:34:32+02:00 INFO [firewall] removing allowed port 1234...
vpn  | 2023-06-05T01:34:33+02:00 INFO [vpn] starting
vpn  | 2023-06-05T01:34:33+02:00 INFO [firewall] allowing VPN connection...
vpn  | 2023-06-05T01:34:33+02:00 INFO [wireguard] Using available kernelspace implementation
vpn  | 2023-06-05T01:34:33+02:00 INFO [wireguard] Connecting to xx.xx.xx.xx:58237
vpn  | 2023-06-05T01:34:33+02:00 INFO [wireguard] Wireguard is up
vpn  | 2023-06-05T01:34:33+02:00 INFO [firewall] setting allowed input port 1234 through interface tun0...
vpn  | 2023-06-05T01:34:33+02:00 INFO [healthcheck] healthy!
vpn  | 2023-06-05T01:34:33+02:00 INFO [ip getter] Public IP address is xx.xx.xx.xx (xx, xx, xx)
vpn  | 2023-06-05T01:34:56+02:00 INFO [healthcheck] unhealthy: dialing: dial tcp4 1.1.1.1:443: i/o timeout
vpn  | 2023-06-05T01:35:01+02:00 INFO [healthcheck] healthy!
vpn  | 2023-06-05T01:35:09+02:00 INFO [healthcheck] unhealthy: dialing: dial tcp4 1.1.1.1:443: i/o timeout
vpn  | 2023-06-05T01:35:14+02:00 INFO [healthcheck] healthy!
vpn  | 2023-06-05T01:35:22+02:00 INFO [healthcheck] unhealthy: dialing: dial tcp4 1.1.1.1:443: i/o timeout
vpn  | 2023-06-05T01:35:27+02:00 INFO [healthcheck] healthy!
vpn  | 2023-06-05T01:35:40+02:00 INFO [healthcheck] unhealthy: dialing: dial tcp4 1.1.1.1:443: i/o timeout
vpn  | 2023-06-05T01:35:46+02:00 INFO [healthcheck] healthy!
vpn  | 2023-06-05T01:35:59+02:00 INFO [healthcheck] unhealthy: dialing: dial tcp4 1.1.1.1:443: i/o timeout
vpn  | 2023-06-05T01:36:02+02:00 INFO [healthcheck] healthy!
vpn  | 2023-06-05T01:36:15+02:00 INFO [healthcheck] unhealthy: dialing: dial tcp4 1.1.1.1:443: i/o timeout
vpn  | 2023-06-05T01:36:20+02:00 INFO [healthcheck] healthy!
vpn  | 2023-06-05T01:36:28+02:00 INFO [healthcheck] unhealthy: dialing: dial tcp4 1.1.1.1:443: i/o timeout
vpn  | 2023-06-05T01:36:33+02:00 INFO [healthcheck] healthy!
vpn  | 2023-06-05T01:36:41+02:00 INFO [healthcheck] unhealthy: dialing: dial tcp4 1.1.1.1:443: i/o timeout
vpn  | 2023-06-05T01:36:46+02:00 INFO [healthcheck] healthy!
vpn  | 2023-06-05T01:36:59+02:00 INFO [healthcheck] unhealthy: dialing: dial tcp4 1.1.1.1:443: i/o timeout
vpn  | 2023-06-05T01:37:04+02:00 INFO [healthcheck] healthy!
vpn  | 2023-06-05T01:37:12+02:00 INFO [healthcheck] unhealthy: dialing: dial tcp4 1.1.1.1:443: i/o timeout
vpn  | 2023-06-05T01:37:17+02:00 INFO [healthcheck] healthy!
vpn  | 2023-06-05T01:37:30+02:00 INFO [healthcheck] unhealthy: dialing: dial tcp4 1.1.1.1:443: i/o timeout
vpn  | 2023-06-05T01:37:35+02:00 INFO [healthcheck] healthy!
vpn  | 2023-06-05T01:37:43+02:00 INFO [healthcheck] unhealthy: dialing: dial tcp4 1.1.1.1:443: i/o timeout
vpn  | 2023-06-05T01:37:48+02:00 INFO [healthcheck] healthy!
vpn  | 2023-06-05T01:37:56+02:00 INFO [healthcheck] unhealthy: dialing: dial tcp4 1.1.1.1:443: i/o timeout
vpn  | 2023-06-05T01:38:04+02:00 INFO [healthcheck] program has been unhealthy for 6s: restarting VPN (see https://github.com/qdm12/gluetun/wiki/Healthcheck)
vpn  | 2023-06-05T01:38:04+02:00 INFO [vpn] stopping
vpn  | 2023-06-05T01:38:04+02:00 INFO [firewall] removing allowed port 1234...
vpn  | 2023-06-05T01:38:05+02:00 INFO [vpn] starting
vpn  | 2023-06-05T01:38:05+02:00 INFO [firewall] allowing VPN connection...
vpn  | 2023-06-05T01:38:05+02:00 INFO [wireguard] Using available kernelspace implementation
vpn  | 2023-06-05T01:38:05+02:00 INFO [wireguard] Connecting to xx.xx.xx.xx:58237
vpn  | 2023-06-05T01:38:05+02:00 INFO [wireguard] Wireguard is up
vpn  | 2023-06-05T01:38:05+02:00 INFO [firewall] setting allowed input port 1234 through interface tun0...
vpn  | 2023-06-05T01:38:05+02:00 INFO [healthcheck] healthy!
vpn  | 2023-06-05T01:38:06+02:00 INFO [ip getter] Public IP address is x.x.x.x (xx, xx, xx)
vpn  | 2023-06-05T01:38:28+02:00 INFO [healthcheck] unhealthy: dialing: dial tcp4 1.1.1.1:443: i/o timeout
vpn  | 2023-06-05T01:38:31+02:00 INFO [healthcheck] healthy!
vpn  | 2023-06-05T01:38:44+02:00 INFO [healthcheck] unhealthy: dialing: dial tcp4 1.1.1.1:443: i/o timeout
vpn  | 2023-06-05T01:38:49+02:00 INFO [healthcheck] healthy!
vpn  | 2023-06-05T01:38:57+02:00 INFO [healthcheck] unhealthy: dialing: dial tcp4 1.1.1.1:443: i/o timeout
vpn  | 2023-06-05T01:39:02+02:00 INFO [healthcheck] healthy!
vpn  | 2023-06-05T01:39:15+02:00 INFO [healthcheck] unhealthy: dialing: dial tcp4 1.1.1.1:443: i/o timeout
vpn  | 2023-06-05T01:39:21+02:00 INFO [healthcheck] healthy!
vpn  | 2023-06-05T01:39:34+02:00 INFO [healthcheck] unhealthy: dialing: dial tcp4 1.1.1.1:443: i/o timeout
vpn  | 2023-06-05T01:39:38+02:00 INFO [healthcheck] healthy!
vpn  | 2023-06-05T01:39:56+02:00 INFO [healthcheck] unhealthy: dialing: dial tcp4 1.1.1.1:443: i/o timeout
vpn  | 2023-06-05T01:39:59+02:00 INFO [healthcheck] healthy!
vpn  | 2023-06-05T01:40:12+02:00 INFO [healthcheck] unhealthy: dialing: dial tcp4 1.1.1.1:443: i/o timeout
vpn  | 2023-06-05T01:40:18+02:00 INFO [healthcheck] healthy!
```

### Share your configuration

```yml
vpn:
    image: qmcgaw/gluetun
    container_name: vpn
    restart: unless-stopped
    cap_add:
      - NET_ADMIN
    environment:
      - PUID=${UID:?err}
      - PGID=${GID:?err}
      - TZ=${TIMEZONE:?err}
      - VPN_SERVICE_PROVIDER=${GT_VPN_SERVICE_PROVIDER:?err}
      - VPN_TYPE=${GT_VPN_TYPE:?err}
      - WIREGUARD_PRIVATE_KEY=${GT_WIREGUARD_PRIVATE_KEY:?err}
      - WIREGUARD_ADDRESSES=${GT_WIREGUARD_ADDRESSES:?err}
      - FIREWALL_VPN_INPUT_PORTS=${GT_FIREWALL_VPN_INPUT_PORTS:?err}
      #- SERVER_CITIES=${GT_SERVER_CITIES:?err}
      - SERVER_COUNTRIES=${GT_SERVER_COUNTRIES}
      - HTTPPROXY=on
      - HTTPPROXY_STEALTH=on
      - HTTPPROXY_USER=${GT_HTTPPROXY_USER:?err}
      - HTTPPROXY_PASSWORD=${GT_HTTPPROXY_PASSWORD:?err}
      - DOT_PROVIDERS=${GT_DOT_PROVIDERS:?err}
      - DOT_CACHING=${GT_DOT_CACHING:?err}
      - HEALTH_TARGET_ADDRESS=${GT_HEALTH_TARGET_ADDRESS:?err}
      - FIREWALL_OUTBOUND_SUBNETS=${GT_LOCAL_SUBNET:?err}
    sysctls:
      - net.ipv6.conf.all.disable_ipv6=0
```
env
```json
"Env": [
                "PUID=1000",
                "HTTPPROXY=on",
                "HTTPPROXY_STEALTH=on",
                "PGID=1000",
                "WIREGUARD_PRIVATE_KEY=wireguradkeycensored=",
                "SERVER_COUNTRIES=Czech Republic,Germany,Poland",
                "HEALTH_TARGET_ADDRESS=1.1.1.1",
                "VPN_SERVICE_PROVIDER=ivpn",
                "WIREGUARD_ADDRESSES=ivpnipv4censored/32,ivpnip6cesnored/128",
                "DOT_CACHING=off",
                "TZ=Europe/Berlin",
                "FIREWALL_VPN_INPUT_PORTS=1234",
                "DOT_PROVIDERS=quad9",
                "HTTPPROXY_PASSWORD=password",
                "FIREWALL_OUTBOUND_SUBNETS=192.168.1.0/24",
                "HTTPPROXY_USER=dietpi",
                "VPN_TYPE=wireguard",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "VPN_ENDPOINT_IP=",
                "VPN_ENDPOINT_PORT=",
                "VPN_INTERFACE=tun0",
                "OPENVPN_PROTOCOL=udp",
                "OPENVPN_USER=",
                "OPENVPN_PASSWORD=",
                "OPENVPN_USER_SECRETFILE=/run/secrets/openvpn_user",
                "OPENVPN_PASSWORD_SECRETFILE=/run/secrets/openvpn_password",
                "OPENVPN_VERSION=2.5",
                "OPENVPN_VERBOSITY=1",
                "OPENVPN_FLAGS=",
                "OPENVPN_CIPHERS=",
                "OPENVPN_AUTH=",
                "OPENVPN_PROCESS_USER=",
                "OPENVPN_CUSTOM_CONFIG=",
                "WIREGUARD_PRESHARED_KEY=",
                "WIREGUARD_PUBLIC_KEY=",
                "WIREGUARD_MTU=",
                "WIREGUARD_IMPLEMENTATION=auto",
                "SERVER_REGIONS=",
                "SERVER_CITIES=",
                "SERVER_HOSTNAMES=",
                "ISP=",
                "OWNED_ONLY=no",
                "PRIVATE_INTERNET_ACCESS_OPENVPN_ENCRYPTION_PRESET=",
                "VPN_PORT_FORWARDING=off",
                "VPN_PORT_FORWARDING_STATUS_FILE=/tmp/gluetun/forwarded_port",
                "OPENVPN_CERT=",
                "OPENVPN_KEY=",
                "OPENVPN_CLIENTCRT_SECRETFILE=/run/secrets/openvpn_clientcrt",
                "OPENVPN_CLIENTKEY_SECRETFILE=/run/secrets/openvpn_clientkey",
                "OPENVPN_ENCRYPTED_KEY=",
                "OPENVPN_ENCRYPTED_KEY_SECRETFILE=/run/secrets/openvpn_encrypted_key",
                "OPENVPN_KEY_PASSPHRASE=",
                "OPENVPN_KEY_PASSPHRASE_SECRETFILE=/run/secrets/openvpn_key_passphrase",
                "SERVER_NUMBER=",
                "SERVER_NAMES=",
                "FREE_ONLY=",
                "MULTIHOP_ONLY=",
                "PREMIUM_ONLY=",
                "FIREWALL=on",
                "FIREWALL_INPUT_PORTS=",
                "FIREWALL_DEBUG=off",
                "LOG_LEVEL=info",
                "HEALTH_SERVER_ADDRESS=127.0.0.1:9999",
                "HEALTH_SUCCESS_WAIT_DURATION=5s",
                "HEALTH_VPN_DURATION_INITIAL=6s",
                "HEALTH_VPN_DURATION_ADDITION=5s",
                "DOT=on",
                "DOT_PRIVATE_ADDRESS=127.0.0.1/8,10.0.0.0/8,172.16.0.0/12,192.168.0.0/16,169.254.0.0/16,::1/128,fc00::/7,fe80::/10,::ffff:7f00:1/104,::ffff:a00:0/104,::ffff:a9fe:0/112,::ffff:ac10:0/108,::ffff:c0a8:0/112",
                "DOT_VERBOSITY=1",
                "DOT_VERBOSITY_DETAILS=0",
                "DOT_VALIDATION_LOGLEVEL=0",
                "DOT_IPV6=off",
                "BLOCK_MALICIOUS=on",
                "BLOCK_SURVEILLANCE=off",
                "BLOCK_ADS=off",
                "UNBLOCK=",
                "DNS_UPDATE_PERIOD=24h",
                "DNS_ADDRESS=127.0.0.1",
                "DNS_KEEP_NAMESERVER=off",
                "HTTPPROXY_LOG=off",
                "HTTPPROXY_LISTENING_ADDRESS=:8888",
                "HTTPPROXY_USER_SECRETFILE=/run/secrets/httpproxy_user",
                "HTTPPROXY_PASSWORD_SECRETFILE=/run/secrets/httpproxy_password",
                "SHADOWSOCKS=off",
                "SHADOWSOCKS_LOG=off",
                "SHADOWSOCKS_LISTENING_ADDRESS=:8388",
                "SHADOWSOCKS_PASSWORD=",
                "SHADOWSOCKS_PASSWORD_SECRETFILE=/run/secrets/shadowsocks_password",
                "SHADOWSOCKS_CIPHER=chacha20-ietf-poly1305",
                "HTTP_CONTROL_SERVER_ADDRESS=:8000",
                "UPDATER_PERIOD=0",
                "UPDATER_MIN_RATIO=0.8",
                "UPDATER_VPN_SERVICE_PROVIDERS=",
                "PUBLICIP_FILE=/tmp/gluetun/ip",
                "PUBLICIP_PERIOD=12h",
                "PPROF_ENABLED=no",
                "PPROF_BLOCK_PROFILE_RATE=0",
                "PPROF_MUTEX_PROFILE_RATE=0",
                "PPROF_HTTP_SERVER_ADDRESS=:6060",
                "VERSION_INFORMATION=on"
            ]
```

**Comment by Stumblinbear** [comment]
I'm getting this as well, except it never restarts, so the container remains disconnected until I manually restart it. This is occurring within minutes of the container being started.

**Comment by qdm12** [comment]
@Yaikava it's likely the connection `i/o timeout` is not very reliable, so that's one or many of:
- your network
- the connection from your router to the vpn server
- the connection between the vpn server and the health ping address

If this bothers you, you can change the health settings see https://github.com/qdm12/gluetun/wiki/Health-options

@Stumblinbear please create another issue since it's unrelated (here it does internally restart the vpn)

**Comment by smolpaw** [comment]
I was facing the same issue. In my case the solution was to not have the same wireguard conf to be already in use by a different container or someplace else
